infra/gcp/ensure-main-project.sh

```diff
@@ -396,11 +396,36 @@ function ensure_prow_special_cases {
         return 1
     fi
     local project="${1}"
+
+    local bucket principal secret
+
     color 6 "Special case: ensuring k8s-infra-ci-robot-github-token accessible by k8s-infra-prow-build-trusted"
-    local principal secret
     principal="serviceAccount:$(svc_acct_email "k8s-infra-prow-build-trusted" "kubernetes-external-secrets")"
     secret=$(secret_full_name "${project}" "k8s-infra-ci-robot-github-token")
-    ensure_secret_role_binding "${secret}" "${principal}" "roles/secretmanager.secretAccessor"
+    ensure_secret_role_binding "${secret}" "${principal}" "roles/secretmanager.secretAccessor" 2>&1 | indent
+
+    color 6 "Special case: ensuring gs://k8s-metrics-canary exists for gs://k8s-metrics migration"
+    (
+      bucket="gs://k8s-project-metrics"
+      owners="k8s-infra-prow-oncall@kubernetes.io"
+      local old_service_account="triage@k8s-gubernator.iam.gserviceaccount.com"
+
+      ensure_public_gcs_bucket "${project}" "${bucket}"
+      ensure_gcs_bucket_auto_deletion "${bucket}" "365" # match gs://k8s-metrics
+      # GCS admins can admin all GCS buckets
+      empower_gcs_admins "${project}" "${bucket}"
+      # bucket owners can admin this bucket
+      empower_group_to_admin_gcs_bucket "${owners}" "${bucket}"
+      # TODO(spiffxp): copy pasted to flip to ensure_removed when migrated
+      # k8s-prow-builds can write to this bucket
+      principal="serviceAccount:${old_service_account}"
+      ensure_gcs_role_binding "${bucket}" "${principal}" "objectAdmin"
+      ensure_gcs_role_binding "${bucket}" "${principal}" "legacyBucketWriter"
+      # k8s-infra-prow-build-trusted can write to this bucket
+      principal="serviceAccount:$(svc_acc_email "k8s-infra-prow-build-trusted" "k8s-metrics")"
```

Comment:
`k8s-metrics` exists ? Can't find it the list of service accounts: 

```shell
gcloud iam service-accounts list --project k8s-infra-prow-build-trusted
DISPLAY NAME                                      EMAIL                                                                              DISABLED
Used by pods in 'prow-build-trusted' GKE cluster  prow-build-trusted@k8s-infra-prow-build-trusted.iam.gserviceaccount.com            False                            False
gcb-builder                                       gcb-builder@k8s-infra-prow-build-trusted.iam.gserviceaccount.com                   False
Nodes in GKE cluster 'prow-build-trusted'         gke-nodes-prow-build-trusted@k8s-infra-prow-build-trusted.iam.gserviceaccount.com  False
prow-deployer                                     prow-deployer@k8s-infra-prow-build-trusted.iam.gserviceaccount.com                 False
kubernetes-external-secrets                       kubernetes-external-secrets@k8s-infra-prow-build-trusted.iam.gserviceaccount.com   False
```

Reply:
Good catch.  Added service account creation to terraform and SA resource that will be deployed to cluster on merge.
```
$ terraform apply
Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # google_service_account.k8s_metrics_sa will be created
  + resource "google_service_account" "k8s_metrics_sa" {
      + account_id   = "k8s-metrics"
      + display_name = "k8s-metrics"
      + email        = (known after apply)
      + id           = (known after apply)
      + name         = (known after apply)
      + project      = "k8s-infra-prow-build-trusted"
      + unique_id    = (known after apply)
    }

  # google_service_account_iam_policy.k8s_metrics_sa_iam will be created
  + resource "google_service_account_iam_policy" "k8s_metrics_sa_iam" {
      + etag               = (known after apply)
      + id                 = (known after apply)
      + policy_data        = jsonencode(
            {
              + bindings = [
                  + {
                      + members = [
                          + "serviceAccount:k8s-infra-prow-build-trusted.svc.id.goog[test-pods/k8s-metrics]",
                        ]
                      + role    = "roles/iam.workloadIdentityUser"
                    },
                ]
            }
        )
      + service_account_id = (known after apply)
    }

Plan: 2 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

google_service_account.k8s_metrics_sa: Creating...
google_service_account.k8s_metrics_sa: Creation complete after 1s [id=projects/k8s-infra-prow-build-trusted/serviceAccounts/k8s-metrics@k8s-infra-prow-build-trusted.iam.gserviceaccount.com]
google_service_account_iam_policy.k8s_metrics_sa_iam: Creating...
google_service_account_iam_policy.k8s_metrics_sa_iam: Creation complete after 1s [id=projects/k8s-infra-prow-build-trusted/serviceAccounts/k8s-metrics@k8s-infra-prow-build-trusted.iam.gserviceaccount.com]

Apply complete! Resources: 2 added, 0 changed, 0 destroyed.
```
```yaml
# $ gcloud iam service-accounts describe k8s-metrics@k8s-infra-prow-build-trusted.iam.gserviceaccount.com
displayName: k8s-metrics
email: k8s-metrics@k8s-infra-prow-build-trusted.iam.gserviceaccount.com
etag: MDEwMjE5MjA=
name: projects/k8s-infra-prow-build-trusted/serviceAccounts/k8s-metrics@k8s-infra-prow-build-trusted.iam.gserviceaccount.com
oauth2ClientId: '108684425368600202785'
projectId: k8s-infra-prow-build-trusted
uniqueId: '108684425368600202785'
```